### PR TITLE
add trap to fetch hostname given node number

### PR DESCRIPTION
**Pull request by aakash10292**
This trap looks up a hostname given a node number. It also toggles verbose logging just for the duration of trap execution to provide detailed trace.  
This is an effort to understand and fix RDSICDBDBA-1342

### CODE REVIEW

**Review by roborivers — COMMENTED** [review]
Lint check:

```diff


@@ bdb/info.c @@
+ Please make sure that a mutex is unlocked before returning.
```

**Review by roborivers — CHANGES_REQUESTED** [review]
<p><a href='https://bloomberg.github.io/comdb2/contrib.html#code-style'>Coding style</a> check: <b>Error</b>. &#x26A0;.<br>Smoke testing: Success &#10003;.<br>Cbuild submission: Error &#x26A0;.<br>Regression testing: <b>4/554 tests failed</b> &#x26A0;.</p><p>The first 10 failing tests are: <br>sc_resume_logicalsc_generated<br>sc_resume<br>sc_lotsoftables<br>truncatesc_offline_generated</p>

**Review by roborivers — COMMENTED** [review]
Lint check:

```diff


@@ bdb/info.c @@
+ Please make sure that a mutex is unlocked before returning.
```

**Review by roborivers — CHANGES_REQUESTED** [review]
<p><a href='https://bloomberg.github.io/comdb2/contrib.html#code-style'>Coding style</a> check: <b>Error</b>. &#x26A0;.<br>Smoke testing: Success &#10003;.<br>Cbuild submission: Error &#x26A0;.<br>Regression testing: <b>4/554 tests failed</b> &#x26A0;.</p><p>The first 10 failing tests are: <br>sc_transactional_rowlocks_generated<br>sc_downgrade<br>incoh_remsql<br>truncatesc_offline_generated</p>

**Review by roborivers — COMMENTED** [review]
Lint check:

```diff


@@ bdb/info.c @@
+ Please make sure that a mutex is unlocked before returning.
```

**Review by roborivers — CHANGES_REQUESTED** [review]
<p><a href='https://bloomberg.github.io/comdb2/contrib.html#code-style'>Coding style</a> check: <b>Error</b>. &#x26A0;.<br>Smoke testing: Success &#10003;.<br>Cbuild submission: Error &#x26A0;.<br>Regression testing: <b>5/554 tests failed</b> &#x26A0;.</p><p>The first 10 failing tests are: <br>sc_resume_logicalsc_generated<br>logfill_logput_window_generated<br>analyze_partial_index_off_generated<br>sc_downgrade<br>truncatesc_offline_generated</p>

**Review by roborivers — COMMENTED** [review]
Lint check:

```diff


@@ bdb/info.c @@
+ Please make sure that a mutex is unlocked before returning.
```

**Review by roborivers — CHANGES_REQUESTED** [review]
<p><a href='https://bloomberg.github.io/comdb2/contrib.html#code-style'>Coding style</a> check: <b>Error</b>. &#x26A0;.<br>Smoke testing: Success &#10003;.<br>Cbuild submission: Success &#10003;.<br>Regression testing: <b>5/554 tests failed</b> &#x26A0;.</p><p>The first 10 failing tests are: <br>lostwrite [setup failure]<br>sc_resume_logicalsc_generated<br>sc_timepart<br>sc_downgrade<br>truncatesc_offline_generated</p>

File: bdb/info.c [hunk 1]
```diff
@@ -1665,9 +1671,21 @@ void bdb_process_user_command(bdb_state_type *bdb_state, char *line, int lline,
...
-    else if (tokcmp(tok, ltok, "del") == 0) {
+    } else if (tokcmp(tok, ltok, "hostname") == 0) {
+        char *iHost, *oHost;
+        Pthread_mutex_lock(&hostname_lookup_lk);
+        gbl_verbose_hostname_lookup = 1;
+        iHost = segtok(line, lline, &st, &ltok);
+        if (ltok == 0) {
+            logmsg(LOGMSG_USER, "expected hostname\n");
+            Pthread_mutex_unlock(&hostname_lookup_lk);
+            return;
```
Comment (morgando) [inline bdb/info.c:1682]: I think maybe set `gbl_verbose_hostname_lookup` to 0 here.

**Review by morgando — CHANGES_REQUESTED** [review]

File: bdb/info.c (hunk 1 above)
Comment (aakash10292) [reply bdb/info.c:1683]: Done!

**Review by morgando — APPROVED** [review]
Looks good!

File: bdb/info.c (hunk 1 above)
Comment (riverszhang89) [reply bdb/info.c:1683]: It does not seem that the  `gbl_verbose_hostname_lookup `  flag protects anything, though. Do we need the mutex here, too?

**Comment by aakash10292** [comment]
cdb2test   Jun 14 16:55:54 2024  in-comdb2db aar_bdb_hostname.R20240614.5

**Pull request merged**